File: 383-Ransom-Note/solution.py
from collections import defaultdict
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        magazine_counter=defaultdict(int)
        ransom_counter=defaultdict(int)
        for x in ransomNote:
            ransom_counter[x]+=1
        for x in magazine:
            magazine_counter[x]+=1
        for x,count in ransom_counter.iteritems():
            if (magazine_counter[x]<count):
                return False
        return True

File: 383-Ransom-Note/solution.c
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    var randsom_counter={}
    var magazine_counter={}
    for (let i=0;i<ransomNote.length;i++){
        let x=ransomNote[i]
        randsom_counter[x]=(randsom_counter[x]||0)+1
    }
    for (let i=0;i<magazine.length;i++){
        let x=magazine[i]        
        magazine_counter[x]=(magazine_counter[x]||0)+1   
    }
    console.log(ransomNote,magazine,randsom_counter,magazine_counter)
    for (let x in randsom_counter ){
        let count=randsom_counter[x];
        if ((magazine_counter[x]||0)<count)
            return false 
    }
    return true;
    
    
};